### PR TITLE
Player rejoin via cookies

### PR DESCRIPTION
**Pull request by sebcontreras**
In this PR:
- Added cookie logic to rejoin the game as a player
- Updated server url #RIPVercel

Note:
- Cookies should be cleared after the game ends. I'm just waiting for the end game PR to be merged. Once that is merged, I'll add that logic to this PR.

### CODE REVIEW

File: src/components/Timer.tsx [hunk 1]
```diff
@@ -13,25 +13,35 @@ export default function Timer(props: { index: number }) {
...
-            targetTime = timeNow + secondsLeft*1000;
+            const indexCookie = Cookies.get('index');
+            // If we are rejoining as a host OR if we are rejoining as a player at the question we
+            // were at previously, use the stored secondsLeft for the targetTime (coming from the cookies)
+            if (isHost || (!isHost && indexCookie !== undefined && (Number(indexCookie)) === index)) {
+                // targetTime = timeNow + TIME_LIMIT_IN_MS;
+                targetTime = timeNow + secondsLeft * 1000;
+            }
+            dispatch(setRejoinAsHost(false));
+            dispatch(setRejoinAsPlayer(false));
```
Comment (vicphan) [inline src/components/Timer.tsx:36]: Is there a reason why you have this here when you already set these flags to `False` in lines 25-26?

File: src/hooks/usePollForGetQuestion.ts [hunk 2]
```diff
@@ -3,16 +3,20 @@ import axios from "axios";
 import { useEffect, useState } from "react";
 import { API_URL } from "../settings";
 import { setQuestion } from "../state/questionSlice";
-import { setGotQuestion, incrementPollGetQuestionCount, setRequestNextQuestion } from "../state/gameSlice"
+import { setGotQuestion, incrementPollGetQuestionCount, setRequestNextQuestion } from "../state/gameSlice";
+import Cookies from "js-cookie";
```
Comment (vicphan) [inline src/hooks/usePollForGetQuestion.ts:7]:
```suggestion
```

File: src/hooks/usePollForGetQuestion.ts [hunk 3]
```diff
@@ -34,7 +38,7 @@ export function usePollForGetQuestion() {
         const { status, data } = response;
         if (status === 200) {
           console.log("status 200 on usePollForGetQuestion");
-          if (requestNextQuestion && index && (index - 1) === data.index) {
+          if (requestNextQuestion && index && index - 1 === data.index) {
```
Comment (vicphan) [inline src/hooks/usePollForGetQuestion.ts:41]:
```suggestion
          if (requestNextQuestion && index && (index - 1) === data.index) {
```

File: src/hooks/useRejoinAsPlayer.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    const gameCodeCookie = Cookies.get("gameCode");
+    const nameCookie = Cookies.get("name");
+    if (gameCodeCookie !== undefined && nameCookie !== undefined) {
+      dispatch(setIsHost(false));
+      dispatch(setGameCode(gameCodeCookie));
+      dispatch(setPlayerName(nameCookie));
+      dispatch(setGameStarted(true));
+      dispatch(setHasJoinedGame(true));
+      dispatch(resetLeaderboard());
+      dispatch(setRequestUpdatedLeaderboard(false));
```
Comment (vicphan) [inline src/hooks/useRejoinAsPlayer.ts:29]: Don't think these two are needed since the player doesn't see the leaderboard

**Review by vicphan — CHANGES_REQUESTED** [review]
Overall looks good. Just some minor suggestions. But the big thing you'll need to is merge in main to include the stuff that I just worked on (the game ending). I suspect there's gonna be a bunch of merge conflicts, and you'll need to remove the cookies you add here.

**Review by vicphan — COMMENTED** [review]
I also noticed another issue when I started testing. When I'm testing with two different tabs on the same browser (one as host, and another as a player), I noticed that once I start the game, the host suddenly becomes the player. Sometimes if you refresh both pages, both tabs can suddenly become the host. This is probably not gonna come up since the same browser will not likely be both the host and the client but I should let you know. It might be a good idea to address this anyway, maybe storing the name along with the `isHost` cookie and using both to determine whether or not it is the host.

File: src/hooks/usePollForGetQuestion.ts (hunk 3 above)
Comment (sebcontreras) [reply src/hooks/usePollForGetQuestion.ts:41]: https://assets.change.org/photos/8/jw/ax/QMjwAxeQAfcpoxs-800x450-noPad.jpg?1597557421

Comment (sebcontreras) [reply src/hooks/usePollForGetQuestion.ts:41]: i actually tried adding this and it looks like prettier doesn't like it so it removes the brackets

File: src/hooks/useRejoinAsPlayer.ts (hunk 4 above)
Comment (sebcontreras) [reply src/hooks/useRejoinAsPlayer.ts:29]: gotcha good catch

**Comment by sebcontreras** [comment]
> I also noticed another issue when I started testing. When I'm testing with two different tabs on the same browser (one as host, and another as a player), I noticed that once I start the game, the host suddenly becomes the player. Sometimes if you refresh both pages, both tabs can suddenly become the host. This is probably not gonna come up since the same browser will not likely be both the host and the client but I should let you know. It might be a good idea to address this anyway, maybe storing the name along with the `isHost` cookie and using both to determine whether or not it is the host.

Yeah I mentioned this behavior in the [rejoin as host PR](https://github.com/Jhappy77/cpsc559-project-front-end/pull/60). If you try to play as both a client and a host on the same browser then yeah the logic can get real wonky as the cookies start overwriting each other. The cookies are stored in the browser so you can't have an instance of a player and a host on the same browser. Ideally there should be no scenario where a user is allowed to be both a host and a client. The current setup makes these mutually exclusive in a browser. If a user is a host and then joins the same game as a player, they are not longer able to rejoin as hosts and must continue as a player, and vice versa.

I think this behavior is fine specially at this stage of the project. To test this I've been using two different browsers with one as a host and one as a player. This seems to work fine from my testing and simulates two users on different machines.

**Review by vicphan — APPROVED** [review]

**Pull request merged**